RFC0015.md

```diff
@@ -0,0 +1,164 @@
+```
+The _targets_ array identifies the processes to which the requested job control action is to be applied. A NULL value can be used to indicate all processes in the caller's nspace. The use of _PMIX\_RANK\_WILDARD_ in a target _pmix\_proc\_t_ can also be used to indicate that all processes in the given nspace are to be included.
+
+The callback function provides a status to indicate whether or not the request was granted, and to provide some information as to the reason for any denial in the _pmix\_info\_cbfunc\_t_ array of _pmix\_info\_t_ structures.
+
+Currently identified attributes (and their associated data type) that can be used to steer behavior via the directives array include:
+```c
+/* job control attributes */
+#define PMIX_JOB_CTRL_ID                    "pmix.jctrl.id"         // (char*) provide a string identifier for this request
+#define PMIX_JOB_CTRL_PAUSE                 "pmix.jctrl.pause"      // (bool) pause the specified processes
+#define PMIX_JOB_CTRL_RESUME                "pmix.jctrl.resume"     // (bool) "un-pause" the specified processes
+#define PMIX_JOB_CTRL_CANCEL                "pmix.jctrl.cancel"     // (char*) cancel the specified request
+                                                                    //         (NULL => cancel all requests from this requestor)
+#define PMIX_JOB_CTRL_KILL                  "pmix.jctrl.kill"       // (bool) forcibly terminate the specified processes and cleanup
+#define PMIX_JOB_CTRL_RESTART               "pmix.jctrl.restart"    // (char*) restart the specified processes using the given checkpoint ID
+#define PMIX_JOB_CTRL_CHECKPOINT            "pmix.jctrl.ckpt"       // (char*) checkpoint the specified processes and assign the given ID to it
+#define PMIX_JOB_CTRL_CHECKPOINT_EVENT      "pmix.jctrl.ckptev"     // (bool) use event notification to trigger process checkpoint
+#define PMIX_JOB_CTRL_CHECKPOINT_SIGNAL     "pmix.jctrl.ckptsig"    // (int) use the given signal to trigger process checkpoint
+#define PMIX_JOB_CTRL_CHECKPOINT_TIMEOUT    "pmix.jctrl.ckptsig"    // (int) time in seconds to wait for checkpoint to complete
+#define PMIX_JOB_CTRL_SIGNAL                "pmix.jctrl.sig"        // (int) send given signal to specified processes
+#define PMIX_JOB_CTRL_PROVISION             "pmix.jctrl.pvn"        // (char*) regex identifying nodes that are to be provisioned
+#define PMIX_JOB_CTRL_PROVISION_IMAGE       "pmix.jctrl.pvnimg"     // (char*) name of the image that is to be provisioned
+#define PMIX_JOB_CTRL_PREEMPTIBLE           "pmix.jctrl.preempt"    // (bool) declare that job can be pre-empted
+```
+Requests use arrays of attributes to fully describe the request - e.g., the directives array might contain the following:
+```
+pmix_info_t directives[2];
+PMIX_INFO_LOAD(&directives[0], PMIX_JOB_CTRL_CHECKPOINT, "mycheckpoint.1", PMIX_STRING)
+PMIX_INFO_LOAD(&directives[1], PMIX_JOB_CTRL_CHECKPOINT_SIGNAL, SIGUSR2, PMIX_INT);
+```
```

Comment:
This example makes me think about the SIGNAL/EVENT. 

In this example, the requestor sets that a given signal should be triggered at the target clients. However, if the requestor is a tool, and the client an application process, the tool may manage a variety of application client types that employ a checkpointing system of their liking. It seems to me that in such cases, the target client should publish to its local server what is the signal/event that should be used to trigger a checkpoint, which would leave the requestor free of knowing what particular signal that client needs to trigger.